dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,313 @@
+	}{
+		{
+			name: "base case",
+			disjointBlockSet: disjointBlockSet{
+				limit: 0,
+				blocks: map[common.Hash]*pendingBlock{
+					{1}: testBlock,
+				},
+				timeNow: mockTime,
+			},
+			remaining: 0,
+		},
+		{
+			name: "remove clear one block",
+			disjointBlockSet: disjointBlockSet{
+				limit: 0,
+				blocks: map[common.Hash]*pendingBlock{
+					{1}: testBlock,
+					{2}: testBlock2,
+				},
+				timeNow: mockTime,
+			},
+			remaining: 1,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.disjointBlockSet.clearBlocks()
+			assert.Equal(t, tt.remaining, len(tt.disjointBlockSet.blocks))
+		})
+	}
+}
+
+func Test_disjointBlockSet_getBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock := &pendingBlock{}
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		want             []*pendingBlock
+	}{
+		{
+			name: "no blocks",
+			want: []*pendingBlock{},
+		},
+		{
+			name: "base case",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{}: testBlock,
+				},
+			},
+			want: []*pendingBlock{testBlock},
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &disjointBlockSet{
+				blocks: tt.disjointBlockSet.blocks,
+			}
+			blocks := s.getBlocks()
+			assert.Equal(t, tt.want, blocks)
+		})
+	}
+}
+
+func Test_disjointBlockSet_removeLowerBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock1 := &pendingBlock{
+		hash:   common.Hash{1},
+		number: 1,
+	}
+	testBlock10 := &pendingBlock{
+		hash:   common.Hash{10},
+		number: 10,
+	}
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		num              uint
+		remaining        int
+	}{
+		{
+			name: "number 0",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       0,
+			remaining: 2,
+		},
+		{
+			name: "number 1",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       1,
+			remaining: 1,
+		},
+		{
+			name: "number 11",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       11,
+			remaining: 0,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.disjointBlockSet.removeLowerBlocks(tt.num)
+			assert.Equal(t, tt.remaining, len(tt.disjointBlockSet.blocks))
+		})
+	}
+}
+
+func Test_disjointBlockSet_size(t *testing.T) {
+	t.Parallel()
+
+	testBlock1 := &pendingBlock{
+		hash:   common.Hash{1},
+		number: 1,
+	}
+	testBlock10 := &pendingBlock{
+		hash:   common.Hash{10},
+		number: 10,
+	}
```

Comment:
nit you could define these inline to have more local test case data. You can also use short syntax like

```go
{10}: {hash: common.Hash{10}, number: 10}
```

which is not too long

Reply:
Updated.

Reply:
I don't think the changes are pushed 🤔